workflows/phaseimpute/tests/main.nf.test

```diff
@@ -24,7 +24,8 @@ nextflow_pipeline {
                         .list()
                         .collect { getRecursiveFileNames(it, outputDir) }
                         .flatten(),
-                    path("$outputDir/imputation/glimpse1/concat/NA12878.glimpse1.vcf.gz").vcf.summary
+                    path("$outputDir/imputation/glimpse1/concat/all_batch0.glimpse1.vcf.gz").vcf.summary.replaceAll(", phasedAutodetect=(false|true)", ""),
```

Comment:
The validation step can manage multiple samples in the vcf.
We have two solution here:

- Split the vcf per samples (as done with stitch before)
- Aggregate everything in one VCF for all batch files

Which would you prefer ?

Reply:
Splitting per samples might take more space and also add another step.
We could add a parameter for that such as `--split_imputed_by <sample,batch,none>`. This would split or not the vcf per tools as follow:

- `sample` each sample has its own vcf
- `batch` the imputed vcf contains all the individuals in each batch
- `none` all the samples in one big vcf

This will also impact the validation process. You will obtain a validation process by resulting vcf, allowing finer detailed validation if `sample` is used and more global if `none`.

Reply:
I would approach this differently.
If `batch_size = 0` (the default, no batches) then the samples should be processed as before and the output should be the exact same as before. 1 vcf per sample. All tools can handle this. 
If `batch_size > 0`, then the output can be a multisample VCF.

My thoughts come for the following: the user is using as input _individual_ bam files, so I would expect the output to be individual VCFs. 

In this way, the previous tests should behave exactly as in the past and produce the same output files/md5sum. 
In the batch test, as we are explicitly asking for batch processing, then the output could be a multisample VCF **and** optionally single-sample VCFs upon request.

Reply:
The only issue is that stitch and glimpse1 should impute in batch (stitch was already doing that previously).
I've explained it more in the usage.md why we should impute in batch:

> Imputation softwares algorithm are time consuming. The computational load depend on the number of individuals, the region size and the panel size. [Some steps are computationally fixed](https://doi.org/10.1038/s41588-023-01438-3), meaning they run similarly whether you are imputing 2 individuals or 200. By grouping individuals into larger batches, these fixed-cost steps are shared among more samples, reducing the per-individual computational overhead and improving overall efficiency. This step is recommended
On the other hand we also need to limit the memory usage when working with a huge amount of individuals within a process.
Hence the necessity to use a batch_size large enough to reduce the fixed-cost stepts / individuals and not to large for the memory usage to be sustainable.

I agree on the other hand that the resulting vcf should be by sample.
This could be achieve by adding a vcf_split_bcftools after each imputation (this would make it easy also for the validation).